compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -266,6 +266,32 @@ impl DataFlowGraph {
         self.intrinsics.get(&intrinsic)
     }
 
+    pub(crate) fn get_intrinsic_from_value(&self, value: ValueId) -> Option<Intrinsic> {
+        let intrinsics = self
+            .intrinsics
+            .iter()
+            .filter_map(
+                |(intrinsic, intrinsic_value_id)| {
```

Comment:
If you have a ValueId that refers to an intrinsic you should be able to retrieve it via:
```rs
match &dfg[value] {
    Value::Intrinsic(foo) => ....,
    _ => (),
}
```

Reply:
Ah yes, done